File: src/tests/test_parser/MemberAccessing.java
public class Test {
  void g() {
    this.foo.x += 1;
  }
  void fake() {
    foo(123);
  }
  void g2() {
    this.foo.x++;
  }
  void g3() {
    this.fun(this.foo);
  }
  void g4() {
    fun(foo);
  }
  void g5() {
    fun(foo.abc);
  }
  something foo;
}

File: src/tests/test_parser/MemberAccessing.py
from JavaParserVisitor import ClassInfo, MethodInfo

def expectedAnswer():
    ans = dict()
    mainClassInfo = ClassInfo("Test")
    mainClassInfo.members.add("foo")
    mainClassInfo.methods["fake"] = MethodInfo("fake")
    
    for meth in ["g", "g2", "g3", "g4", "g5"]:
        method = MethodInfo(meth)
        method.memberReferences.add("foo")
        mainClassInfo.methods[meth] = method
    
    ans["Test"] = mainClassInfo
    
    return ans